### ISSUE TITLE
Root `+server.js` that is prerendered is always saved as `index.html`

### ISSUE THREAD

**Issue by eltigerchino** [sveltejs/kit#11087]
### Describe the bug

Even when `is_html` is false (e.g., content-type is not text/html), the prerendered root endpoint (src/routes/+server.js) gets saved as `index.html`. This is caused by `path.slice(config.paths.base.length + 1)` returning an empty string which is falsy, so the `'index.html` filename gets returned.

https://github.com/sveltejs/kit/blob/4559e66b649516cc996a042a21739fa165388335/packages/kit/src/core/postbuild/prerender.js#L145-L153

### Reproduction

n/a

### System Info

```Shell
n/a
```

### Severity

serious, but I can work around it

**Comment by krish-italiya** [comment]
Is the issue still open ?
If yes can I solve it ?

**Comment by geodask** [comment]
No recent updates here, I’d be glad to take this on. The root path is a bit of a puzzle since `/index.html` is typically served for `/`, but with a prerendered `+server.js`, that default might not suit. What should the root file be named in this case to ensure it’s still accessible when visiting `/`?

**Comment by aewing** [comment]
EDIT: What would we expect to happen here? index.[content-type file extension], i.e. index.json?

Static sites wouldn't really know what to do with a non-html static base file, but I guess that's something the user would already have in mind if they needed this functionality.

@eltigerchino

**Comment by teemingc** [comment]
> EDIT: What would we expect to happen here? index.[content-type file extension], i.e. index.json?
> 
> Static sites wouldn't really know what to do with a non-html static base file, but I guess that's something the user would already have in mind if they needed this functionality.
> 
> [@eltigerchino](https://github.com/eltigerchino)

Yeah, sorry this was a poorly filed issue. It was a while ago but I think this stemmed from someone's reproduction where they had no root page but they had an endpoint in the root instead and the rest of their app in a subroute such as `/admin/*`. While completely valid, I think it's definitely an edge case since most people wouldn't do this. However, it just doesn't work at the moment due to the issue mentioned earlier.

**Comment by aewing** [comment]
I implemented a fix in draft right now for this.
The approach I took was to map content-type to extension with a simple heuristic on root-scoped prerender with an endpoint, so for application/json we generate index.json and warn the user about non-standard index files and proper prerender usage.

**Comment by teemingc** [comment]
> I would accept a PR that threw an error if a prerenderable root +server.js responded with a non-HTML response.

See https://github.com/sveltejs/kit/pull/14135#issuecomment-3165007509

**Comment by geodask** [comment]
@teemingc I have opened a PR addressing this issue
